pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -26,72 +26,94 @@ import (
 
 func (c *Cluster) prepareStorageClassDeviceSets(config *provisionConfig) []rookalpha.VolumeSource {
 	volumeSources := []rookalpha.VolumeSource{}
+
+	// Iterate over storageClassDeviceSet
 	for _, storageClassDeviceSet := range c.DesiredStorage.StorageClassDeviceSets {
 		if err := opspec.CheckPodMemory(storageClassDeviceSet.Resources, cephOsdPodMinimumMemory); err != nil {
-			config.addError("cannot use storageClassDeviceSet %s for creating osds %v", storageClassDeviceSet.Name, err)
+			config.addError("cannot use storageClassDeviceSet %q for creating osds %v", storageClassDeviceSet.Name, err)
 			continue
 		}
 		for i := 0; i < storageClassDeviceSet.Count; i++ {
-			pvc, err := c.createStorageClassDeviceSetPVC(storageClassDeviceSet, i)
-			if err != nil {
-				config.addError("%v", err)
-				config.addError("OSD creation for storageClassDeviceSet %v failed for count %v", storageClassDeviceSet.Name, i)
-				continue
+			// Check if the volume claim template has PVCs
+			if len(storageClassDeviceSet.VolumeClaimTemplates) == 0 {
+				config.addError("no PVC available for storageClassDeviceSet %q", storageClassDeviceSet.Name)
+				return nil
```

Comment:
how about `continue` instead of `return`?

Reply:
seems ok